src/ray/gcs/gcs_server/gcs_task_manager.cc

```diff
@@ -149,6 +149,20 @@ void GcsTaskManager::GcsTaskManagerStorage::UpdateExistingTaskAttempt(
   // Update the task event.
   existing_task.MergeFrom(task_events);
 
+  // Truncate the profile events if needed.
+  auto max_num_profile_events_per_task =
+      RayConfig::instance().task_events_max_num_profile_events_per_task();
+  if (existing_task.profile_events().events_size() > max_num_profile_events_per_task) {
+    auto to_drop =
+        existing_task.profile_events().events_size() - max_num_profile_events_per_task;
+    existing_task.mutable_profile_events()->mutable_events()->DeleteSubrange(0, to_drop);
```

Comment:
if to_drop is negative, it will be noop?

Reply:
it will not enter the if clause in this case.